### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by Nillic** [zippity21/Stardew_Valley_Regression_Mod#102] [General]
Hello, first all thanks for the work on this mod and keeping it alive. Reading the discussions on here seemed like you wanted for feedback so here I am.

- I'm going to start with the elephant in the room : I think the kids dialogue is out of the line. It should be a tell that it felt needed to add a content warning for it in a mod that doesn't do it for scat. I don't really like ABDL as a term but it does mean adult, Jas and Vincent are children. Putting them in diapers in that context involves them with the adult related part of the mod and this is sure to cause troubles at some point. I don't think it's problem if they react to whatever is happening to the player but making both the youngest children in town magically potty untrained is tendancious, even with the best intentions. I may make it sound worse than I think it is, but it shouldn't be in the base mod. I don't think it's bad bad, I wouldn't even be here if I though it was, but it's for sure out of touch.

- Overall, is having dialogue part of the main mod necessary anyway ? Content patcher makes editing dialogue very simple so having a default very neutral or even blank dialogue/event asset that anyone could modify should work to make to custom the general tone of the game without ressorting to coding anything. Anything that anyone have written could simply be copy pasted arround without even needing to make a mod out of it since it's basically just a text. Currently, tone was changed from being very centered arround loss of control to being very centered arround baby play and there's bound to be people that prefer one or the other. Content Patcher is somewhat limited, but it's very simple to use and flexible enough to have fun with and let anyone chose what they want.

- What's makes this mod interesting to me is the more immersive part of the gameplay and how it goes well with the progressive aspect of the mode. The base game have no standard defined progression but function more on a day to day basics. Because of that, I think it's better not to shove kinky content as fast as possible than the opposite. If you just get your initial diaper stock and then mash F1 until you lose complete bladde control, there's nothing left to do outside playing regular Stardew Valley untill you get a high relationship with someone to maybe trigger a new reaction dialogue. I think it would be better to make make the mod slower. For exemple, instead of continence being linear like it is right now it could be decreasing slower and increasing faster at high continence and doing the opposite at low continence. There's also a function for failing to wet or mess in the code but from what I've seen it's mostly limited if your urge is very low, isn't it possible to use it to make that higher continence makes it harder to wet or mess outside toilet (especially doing it on prupose) ? With a changes like this, it's possible to simulate an actual potty training.
There's also something to do with access to diapers (it's very strange that Pierre is selling most of them, the mall is selling less and the clinic is selling none) and Harvey and the clinic having stricly no interaction with the mode despite the theme. Also kinda weird Linus would judge the player harshly for relieving themself outdoor.

- There was discussion about difficulty at some point and as someone who discovered the game, hunger is quite punishing. The energy penalty from hit makes it hard get anything accomplished once you get too low since energy will completly deplete by doing minimum task (in my case, I could juste water the crops, summer rain saved the day). The main reason is basic ingredient calory gain is very low outside spring onion. It seems that fishing is more efficient to get better food, but's it's strange the game would be punishing for player that don't really understand the game and would default to farm because it's what the base game tells you to do. I think this problem can be solved by increasing the calory you get from less nourishing food but also increasing how fast hunger builds. That way you could eat enough to to do your work without being immediatly exhausted while not making huner a non-factor.

And that's already a lot. I hope that this actually helpfull at least.
I'll probably try poking arround to see what can be done with content patcher but i can't be sure how long I'll keep interest in this. Anyway, thanks again to everyone working on this.

**Comment by TikkuTokku** [comment]
If you're interested in writing dialogue for a character, please let me know! I have set aside archetypes for each potential partner, and it's a whole lot of work doing just the text for one character. I would love help with writing, especially the "meaner" archetypes that I know other people would like a lot even though they don't strike my interest as much.

**Reply by Nillic** [reply]
As I said, I don't think writting dialogue isn't going to do that much for now. It's indeed a lot of work and for what I've seen for now, getting to said dialogue takes a lot of time to get something that is very hit or miss. Espcially when it's very uninteractive like it is right now, where character just move arround one archetype and that's it. If I were to do dialogue I would probably mostly doing something I'm interested in over writting someone else dialogue.

For now I'm mostly trying to see what works and what doesn't. I already managed to get some event running (I trying to trick the game into changing the player, but it didn't work) and did some sprite editing. I guess it's only playing arround until diaper as equipement is a thing but I'm also curious about SMAPI stuff, mostly using conditions to have control over what content patcher does. If there's a way to use underwear and wet/mess statut as conditions for text editing it open for much better text editing.

**Reply by zippity21** [reply]
@Nillic Right now the SMAPI part of the mod can't really interact with the Content Patcher part since the way custom items are handled by the game is poor. It's not hard to create a new crop or variation of existing items, but items with entirely new functionality (like diapers) don't mesh well with the base game mechanics. I'm currently working on a side-branch where I plan on making 'dummy' versions of the items with content patcher (so they'll be able to be used/referenced like regular in-game items), but having them internally converted to the custom item (by capturing the `ReceiveInventoryChangedEven`). Unfortunately, I just haven't had much time to work on it.

**Reply by Nillic** [reply]
Ok, I hoped it was at least possible to grab de the same token that change text dynamically but I guess even this is out of reach. But then using variable like continence level is also out of question ? At least, unless it uses at least another API based mod ? I'm still storta trying to figure this out, I hope I make sense.

I was planing to make so makeshift diaper equipement at some point, mainly to experiment arround conditional editing but it's pretty simple so I don't how usefull it can be.

**Comment by zippity21** [comment]
Thank you for the feedback, its what lets us move towards a better end result.

1. For the kid dialogue, there is a warning on first interaction that allows you to opt-out. Though I do understand the concern. It is kind of a fuzzy area. Most of the dialogue has been graciously written by @TikkuTokku . I've been pretty up front since the beginning that I don't really have a voice on the creative side of the mod; I'm only personally interested in maintaining and enhancing the technical aspect.

1a. Previously, the dialogue was hard-coded into the mod, but is now done through the content patcher, so it should be pretty easy to modify. I've considered bundling different versions of dialogue, since not everybody likes the tone, but somebody would have to write it (and I'm a terrible writer). I do believe having it bundled by default does match the intent of the mod, but I don't believe there are any interdependencies that prohibit someone from removing it altogether. 

2. Regarding 'speed-running' incontinence, I've had this thought/observation before, but ultimately I decided to leave it in the hands of the player; deciding when/how is pretty much based on the honor system and to add obstacles would only add complexity for marginal benefit. I like the idea of non-linear gain/loss, so that may be something to consider. Currently, the idea of it becoming easier/harder comes from the notification of when you need to go; lower continence leads to those messages sometimes not appearing.

2a. I agree that where the diapers are sold don't really make sense. That's pretty easy to change though, if you want to propose a distribution.

3. Yeah, the difficulty balancing has been, well... difficult. I landed on a system of calories since it just made some sort of sense instead of being completely arbitrary and there has been an ongoing effort to assign realistic(ish) values to the different foods. I could possibly add a scaling factor that could be edited in the config file, if you think that could help?

**Reply by Nillic** [reply]
1. The warning nearly makes it worse. It's a warning about children in a fetish oriented mod, I'm not sure I really should be arguing about this. Exactly for the sake of maintaining this, you should be carefull about something that could easily get this mod removed from here. I'm pretty sure most people are not comfortable with what's in there and even using the in-game option still leave you with a bunch of useless lines that are basically modded but vanilla. Even worse, you can't just remove the kid dialogue since the warning is a check to hide the lines and not show it, so you have to manually remove every line about them in the entire mod. This is enough to drive people in the best cases.

2. Yeah the way the notification system evolve is a good touch. I think what I had in mind was also the mail you get at the start of the game and how you're instantly flagged as having troubles and given a bunch of supplies (btw, I'm pretty sure farmer grandpa is dead, so I'm not sure how Jodi got the new from him) it kinda breaks immersion. If she just gave money you could still immediatly use it for diapers or get a more comfortable start. For where you could buy the hospital being unused is a bit sad, they could sell the more tames ones (lavender, training pants and baby) while Joja sell most underwear, their own brand and the cloth diapers. The prenium feel like they should be a bit special, is it possible to make them available in the traveling cart or is too much of a hassle ? If it'ss too anoying, Joja is probably what makes the most sense.

3. I'm not to found of trying to solve an issue like this with an option. Especially in that game since difficulty can vary for one person in the same playthrough. It could maybe work to make it harder, and even that probably would just reduce the income, but to adjust difficulty it wouldn't change a lot. There's already ways to make it easier or just remove hunger if you don't want to deal with it. That's why I was thinking of getting more food from some ingredients. For exemple tomatoes gain is pretty low, I would them to get enough stamina in hope to fish algae at some point, so there's still some leeway to make the game smoother without enabling some wild flower, raw potato eating caveman behavior. It's less of a problem once you can upgrade the house, so just giving a little boost to some early food could be good. One of the forageable, a few farmable one that make sense to be somehwta nourishing on its own. Making another spring onion would be overkill though.

**Reply by TikkuTokku** [reply]
1. Nothing inappropriate happens with the kids. It's just silly and cute. I'm aware that still makes people uncomfortable, so I figured out a way to give the option to just not see it. If you don't like it, just hit the no button, and it's gone.

**Reply by TikkuTokku** [reply]
3. My trick has been crab pots and turn the fish into sashimi. It has been very effective, and it's been cool to have crab pots be useful since I normally don't bother with them.

**Reply by Nillic** [reply]
If it makes people uncomfortable it's probably it's not just silly and cute. Involving children with ABDL stuff means involving them with adult themed erotic content. And it's not just a bit, there's something close to 3 or 4 times the amount of lines about those kids wearings diapers in the current mod than they have in the entire vanilla game. Some of those lines talk about checking their peeking underwear. Let's be real, this isn't a mod for silly and cute things.

There's no reason to defend this, especially since it was already written over something else.

**Reply by TikkuTokku** [reply]
Didn't you say that you think it's okay for them to react to your diapers? That is also involving them in this kind of content, and there are a lot of people who would say you are crossing the line for the same reason you say I did.

Nothing inappropriate happens with the kids, and if that's still not enough for you, I added the option to opt out entirely. 

You also mentioned wanting to write content that specifically matches your interest? What would you like to see in the game? I'm almost done working on Harvey as a soft dom (clearly the content I like), but I've set aside other characters to be more bratty, or allow you to be bratty, or stuff like that. If that's the sort of thing you're interested in having in the game I would love your help adding it in.

**Reply by Nillic** [reply]
I'm also fine with them not commentating at all, I said that as concession. Mostly meant as few comment here and there, just like Jas vanilla "You smell grass..." line. As for immersion it does makes sense they would say something but that's it, but I can't say I would miss it if it's not there. Not doing any change to unerage in this kinda of mod is a pretty sane baseline.
And the point isn't how it is for me, but how it's uploaded in this site and what every one trying the mod is going to get.

And If I were to write something it would not be what you're describing. I'm not too found of "everything is about the kink" type so I would more go from what the base game is building (not a lot from what I saw until now) and see what kind of relationship you. But I'm first intérested to see how deep you can go in branching dialogue and events.

**Comment by Gooberology** [comment]
I dont really think this mod has any 'kinky' or 'adult' related stuff in it. Half the reason i enjoy it is because it focuses on age perception and isnt a blatant fetish sim, and id honestly prefer for it to stay that way.

**Reply by Nillic** [reply]
Matter of perspective I guess, the current text makes feel the exact opposite of what you're describing.

It's a very leaning extremly heavily on infantilisation in a way that is extremly forcefull and do feel like a fetish sim. Looking at the way it was edited over time, it's much more heavy handed than the original. This is exactly why I was advocating to have dialogue seprated. Currently if you remove regression from regression dialogue manifest and launch it works fine. That means even if you separate both, all you need to do is keep your version of the EN.json and everything is still there. This is your own separated baby play mod and you like it, good. Can you also let other people have what they want too ? Especially when it's sensitive enough to warrant a warning.

I do like a bit of regression here and there, the current mod is going way too hard, way too far, way to often. That type of "everything have to do with the kink" writting is not my type and there's not much more to it. I'm not advocating it re-written again in something else, I'm saying it's okay if it the main mode is more neutral in tone and focus more specifically on the system side of things.

**Reply by Gooberology** [reply]
The parts where vincent goes out of his way to grab your attention to do something baby related, maybe. But generally how the two feel about you along with how penny grows to feel about you feels more along the lines of a very gentle "you just fit in more as a kid than an adult, and thats okay" rather than anything particularly fetishistic.

**Reply by Nillic** [reply]
Most of the dialogue isn't a bout this, It's about them wearing diapers and using them. This is the one main common point of absolutely every $p202 flagged line (by the way, it would be better to do other way arround, flagging line with $p201<modded text>|<vanilla text>). And again, I don't like ABDL I don't use this term but it's was used to define the dialogue content and it means adult baby, 10 years old kids are neither adult or babies, you don't make children "do something baby related" in the context of an age play related game and then try arguing is more pure.

And once again, is about make it separate, not remove it from the surface of the earth. There's no need to be this much on the defensive.

**Reply by Gooberology** [reply]
Not really being defensive, moreso I think im misunderstanding what youre talking about.

**Reply by Nillic** [reply]
The dialogue Tikku Tokku is writting currently is a separate mod. It works on its own because it only uses content patcher. So there's no need to ship it with regression. The only thing it needs is changing the EN.json and that's it. Instead of having a code that puts a warning that flag specific lines to remplace them with nothing, just make it a seperate mod specifically for ageplay and you won't even need rack your mind to add things you don't like in it for the sake of covering for what everyone want. It could very easily work that way, nobody will miss anything.

Maybe you don't like to hear that the thing you like is not comfortable for some people but realize that's also what as been done to the original mod author without letting a chance for them to have an opinion about it. I'm sorry I'm really not seeing any good reason to keep the mod that way.
